### ISSUE TITLE
[BUG] Scheduler should consider gradient accumulation while assigning `epoch_steps`?

### ISSUE THREAD

**Issue by rohitgr7** [h2oai/h2o-llmstudio#663]
### 🐛 Bug

Here: https://github.com/h2oai/h2o-llmstudio/blob/a9d72fffef370c7fd0dd9f29ece06ab45b6c5815/train.py#L562

Let's say total data batches are 160 and gradient accumulation is 10. Optimization step is happening only 10 times.
But here scheduler is called every time: https://github.com/h2oai/h2o-llmstudio/blob/a9d72fffef370c7fd0dd9f29ece06ab45b6c5815/train.py#L315-L316

which can lead to
https://discuss.pytorch.org/t/userwarning-detected-call-of-lr-scheduler-step-before-optimizer-step-in-pytorch-1-1-0-and-later-you-should-call-them-in-the-opposite-order-optimizer-step-before-lr-scheduler-step/88295
